specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabric/preview/2022-10-01-preview/managedcluster.json

```diff
@@ -0,0 +1,1980 @@
+        "addonFeatures": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManagedClusterAddOnFeature"
+          },
+          "description": "List of add-on features to enable on the cluster."
+        },
+        "enableAutoOSUpgrade": {
+          "type": "boolean",
+          "description": "Setting this to true enables automatic OS upgrade for the node types that are created using any platform OS image with version 'latest'. The default value for this setting is false."
+        },
+        "zonalResiliency": {
+          "type": "boolean",
+          "description": "Indicates if the cluster has zone resiliency.",
+          "default": false
+        },
+        "applicationTypeVersionsCleanupPolicy": {
+          "$ref": "#/definitions/ApplicationTypeVersionsCleanupPolicy",
+          "description": "The policy used to clean up unused versions."
+        },
+        "enableIpv6": {
+          "type": "boolean",
+          "description": "Setting this to true creates IPv6 address space for the default VNet used by the cluster. This setting cannot be changed once the cluster is created. The default value for this setting is false."
+        },
+        "subnetId": {
+          "type": "string",
+          "description": "If specified, the node types for the cluster are created in this subnet instead of the default VNet. The **networkSecurityRules** specified for the cluster are also applied to this subnet. This setting cannot be changed once the cluster is created."
+        },
+        "ipTags": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/IPTag"
+          },
+          "x-ms-identifiers": [],
+          "description": "The list of IP tags associated with the default public IP address of the cluster."
+        },
+        "ipv6Address": {
+          "type": "string",
+          "description": "IPv6 address for the cluster if IPv6 is enabled.",
+          "readOnly": true
+        },
+        "enableServicePublicIP": {
+          "type": "boolean",
+          "description": "Setting this to true will link the IPv4 address as the ServicePublicIP of the IPv6 address. It can only be set to True if IPv6 is enabled on the cluster."
+        },
+        "auxiliarySubnets": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Subnet"
+          },
+          "x-ms-identifiers": [],
+          "description": "Auxiliary subnets for the cluster."
+        },
+        "serviceEndpoints": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ServiceEndpoint"
+          },
+          "x-ms-identifiers": [],
+          "description": "Service endpoints for subnets in the cluster."
+        },
+        "fasterZonalUpdates": {
+          "type": "boolean",
+          "description": "Indicates whether to use a maximum of 3 UD per zone instead of 5 for Cross Az Node types for faster deployments."
+        }
```

Comment:
consider modelling this as an enum : 


[https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review)

Replace boolean/switch properties with better enum
A Boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Even if you still believe [True, False] are the correct values for a property, you should use a string enum with values [True, False] instead of boolean. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [True, False, Unknown].

Reply:
ok, thanks fore the suggestion. fixed

Reply:
@raosuhas please take a look again with the change suggested